python/tvm/script/special_stmt.py

```diff
@@ -491,6 +491,22 @@ def var(dtype, span):
         super().__init__(var, def_symbol=True)
 
 
+@register
+class BufferVarDef(SpecialStmt):
```

Comment:
cc @spectrometerHBH for tvmscript changes. This is for distinguishing a normal variable from a buffer variable. The latter one requires storage scope. See the change in `test_tvmscript_roundtrip.py`. The tvm script printer was also updated.